database_admin/migrations/041_partition_packages.up.sql

```diff
@@ -0,0 +1,29 @@
+CREATE TABLE IF NOT EXISTS system_package_v1
+(
+    rh_account_id INT NOT NULL REFERENCES rh_account,
+    system_id     INT NOT NULL,
+    package_id    INT NOT NULL REFERENCES package,
+    -- Use null to represent up-to-date packages
+    update_data   JSONB DEFAULT NULL,
+    latest_evra   TEXT GENERATED ALWAYS AS ( ((update_data ->> -1)::jsonb ->> 'evra')::text) STORED,
+    PRIMARY KEY (rh_account_id, system_id, package_id) INCLUDE (latest_evra)
+) PARTITION BY HASH (rh_account_id);
+
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON system_package_v1 TO evaluator;
+GRANT SELECT, UPDATE, DELETE ON system_package_v1 TO listener;
+GRANT SELECT, UPDATE, DELETE ON system_package_v1 TO manager;
+GRANT SELECT, UPDATE, DELETE ON system_package_v1 TO vmaas_sync;
+
+SELECT create_table_partitions('system_package_v1', 128,
+                               $$WITH (fillfactor = '70', autovacuum_vacuum_scale_factor = '0.05')$$);
+
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO evaluator;
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO listener;
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO manager;
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO vmaas_sync;
+
+INSERT INTO system_package_v1 (
```

Comment:
Is it good idea to do this in one big batch?

Reply:
Fixed